### PR TITLE
Update test_limits.py

### LINKED ISSUE #15984: Value error in limit
`Limit((-x + log(exp(x) + 1))/x, x, oo, dir='-').doit()` raises `ValueError: cannot compute leadterm`

### PR DESCRIPTION
**Pull request by nileshpatra**
ADDED TEST FOR THE ISSUE UNDER @XFAIL

Fixes #15984 
Added extra test in series/test_limits.py under @XFAIL

* series
  * added a new test for failing improper limit

### CODE REVIEW

File: sympy/series/tests/test_limits.py [hunk 1]
```diff
@@ -536,3 +536,6 @@ def test_issue_10102():
 
 def test_issue_14377():
     raises(NotImplementedError, lambda: limit(exp(I*x)*sin(pi*x), x, oo))
+    
+def test_issue_15984():
+	assert Limit((-x + log(exp(x) + 1))/x, x, oo, dir='-').doit() == 0
```
Comment (divyanshu132) [inline sympy/series/tests/test_limits.py]: Instead of tab please use proper spaces.

**Comment by nileshpatra** [comment]
@divyanshu132 , can please review it now , I have done the needful

**Comment by oscargus** [comment]
Usually this type of contribution does not have release note entry. Use `NO ENTRY` instead.

Also, please do not use the GitHub editor to perform changes. The original commit actually fixed the problem, then there has been seven additional commits for fixing things, including maybe 30 hours of test time for solving a problem that can be tested in 10 seconds locally. (`bin/test quality`) (With that said, I also push things with this type of error sometimes, but using a local copy is so worth it in the long run.)

**Comment by nileshpatra** [comment]
@oscargus , so has the problem been already resolved from my end?

**Comment by oscargus** [comment]
Well, it is apparently still not correct syntax:
```
  File "/home/travis/.venv-no-setuptools/lib/python2.7/site-packages/sympy/series/tests/test_limits.py", line 542
    assert limit((-x + log(exp(x) + 1))/x) , x , oo) == 0
                                               ^
SyntaxError: invalid syntax
```

What I am saying is that conceptually, you had the more or less correct solution initially. If you had tried to run it on a local computer a lot of test time and commits had been saved.

**Comment by oscargus** [comment]
There is still a syntax error in it... Count the number of left and right brackets.

**Comment by nileshpatra** [comment]
is it correct now?

**Comment by nileshpatra** [comment]
@oscargus , I would be highly obliged if you could help me make this PR successful (pass all tests). I'm new to it and didn't know that linting could make this difference . Please guide .

**Comment by nileshpatra** [comment]
on testing it locally as suggested , and resolving white space / syntax errors , I am getting this error which I'm unable to rectify :  
```
Traceback (most recent call last):
  File "/home/nilesh/Desktop/MY FILES/sympy_codes/sympy/sympy/series/tests/test_limits.py", line 542, in test_issue_15984
    assert limit((-x + log(exp(x) + 1))/x, x, oo, dir='-') == 0
  File "/home/nilesh/Desktop/MY FILES/sympy_codes/sympy/sympy/series/limits.py", line 82, in limit
    return Limit(e, z, z0, dir).doit(deep=False)
  File "/home/nilesh/Desktop/MY FILES/sympy_codes/sympy/sympy/series/limits.py", line 250, in doit
    r = limit(inve.as_leading_term(u), u, S.Zero, "+")
  File "/home/nilesh/Desktop/MY FILES/sympy_codes/sympy/sympy/core/expr.py", line 2983, in as_leading_term
    obj = self._eval_as_leading_term(x)
  File "/home/nilesh/Desktop/MY FILES/sympy_codes/sympy/sympy/core/mul.py", line 1649, in _eval_as_leading_term
    return self.func(*[t.as_leading_term(x) for t in self.args])
  File "/home/nilesh/Desktop/MY FILES/sympy_codes/sympy/sympy/core/mul.py", line 1649, in <listcomp>
    return self.func(*[t.as_leading_term(x) for t in self.args])
  File "/home/nilesh/Desktop/MY FILES/sympy_codes/sympy/sympy/core/expr.py", line 2983, in as_leading_term
    obj = self._eval_as_leading_term(x)
  File "/home/nilesh/Desktop/MY FILES/sympy_codes/sympy/sympy/core/add.py", line 848, in _eval_as_leading_term
    return old.compute_leading_term(x)
  File "/home/nilesh/Desktop/MY FILES/sympy_codes/sympy/sympy/core/expr.py", line 2946, in compute_leading_term
    s = calculate_series(self, x, d).subs(d, log(x))
  File "/home/nilesh/Desktop/MY FILES/sympy_codes/sympy/sympy/series/gruntz.py", line 464, in calculate_series
    for t in e.lseries(x, logx=logx):
  File "/home/nilesh/Desktop/MY FILES/sympy_codes/sympy/sympy/core/expr.py", line 2756, in yield_lseries
    for si in s:
  File "/home/nilesh/Desktop/MY FILES/sympy_codes/sympy/sympy/core/expr.py", line 2821, in _eval_lseries
    series = self._eval_nseries(x, n=n, logx=logx)
  File "/home/nilesh/Desktop/MY FILES/sympy_codes/sympy/sympy/core/add.py", line 391, in _eval_nseries
    terms = [t.nseries(x, n=n, logx=logx) for t in self.args]
  File "/home/nilesh/Desktop/MY FILES/sympy_codes/sympy/sympy/core/add.py", line 391, in <listcomp>
    terms = [t.nseries(x, n=n, logx=logx) for t in self.args]
  File "/home/nilesh/Desktop/MY FILES/sympy_codes/sympy/sympy/core/expr.py", line 2908, in nseries
    return self._eval_nseries(x, n=n, logx=logx)
  File "/home/nilesh/Desktop/MY FILES/sympy_codes/sympy/sympy/functions/elementary/exponential.py", line 767, in _eval_nseries
    a, b = s.leadterm(x)
  File "/home/nilesh/Desktop/MY FILES/sympy_codes/sympy/sympy/core/expr.py", line 3027, in leadterm
    should have been free of x but got %s""" % (self, x, c)))
ValueError: 
cannot compute leadterm(exp(1/_Dummy_23532) + 1, _Dummy_23532). The
coefficient should have been free of x but got exp(1/_Dummy_23532)

```

I would be obliged if I could be helped

**Comment by oscargus** [comment]
I'm not sure what you want to obtain. You have added a test for a bug in Sympy.

If you put the @XFAIL decorator at it, the tests will at least pass (not this one, but since it is expected the total test will be OK).

However, it will not fix the issue. To actually solve the issue, you will need to debug the code and see where it goes wrong. Probably a good idea is to start with a similar expression and see how that executes and compare where it differs.

I have no insight into this particular code so I cannot help you further. Once you have some idea where things goes wrong there are probably other persons that may provide further insights.

**Comment by nileshpatra** [comment]
@oscargus , thank you so much :)) . I will try your idea to start with a similar expression and debug . Thanks for helping out!!

**Comment by czgdp1807** [comment]
Closing in favour of https://github.com/sympy/sympy/pull/16157
Thanks for the contributions.

**Pull request closed without merging**